File: Python/stack.py
class Stack:
    def __init__(self):
        self.stack = []
    def push(self,x):
        self.stack.append(x)
    def pop(self):
        if self.stack:
            return self.stack.pop()
        else:
            print("Pop operation can't performed on empty stack")
    def traverse(self):
        if not self.stack:
            print("Stack is Empty")
        while self.stack:
            print(self.pop())
    
        
        
stack = Stack()
arr = [22,3,45,6,1,7,8]
stack.push(22)
stack.push(3)
stack.push(45)
stack.push(6)
stack.pop()
stack.pop()
stack.pop()
stack.pop()
stack.pop()
# print(stack.stack)
# stack.traverse()
# stack.pop()
# stack.traverse()

File: stack.c
#include<stdio.h>
#define MAX 5
struct Stack
{
    int stack[MAX];
    int sp;
}s;

void init()
{
    s.sp=-1;
}

int isFull()
{
    if (s.sp==MAX-1)
        return 1;
    else
        return 0;
}

int isEmpty()
{
    if (s.sp==-1)
        return 1;
    else
        return 0;
}

void push(int value)
{
    if (isFull())
        printf("Overflow");
    else
        s.stack[++s.sp]=value;
}

int pop()
{
    if (isEmpty())
        printf("Underflow");
    else
        return (s.stack[s.sp--]);
}

void display()
{
    int tos=s.sp;
    printf("\nStack values \n");
    for (int i=tos;i>-1;i--)
        printf("  %d",s.stack[i]);
}

int main()
{
    int choice;
    init();
    while(1)
    {
        printf("\n1. PUSH");
        printf("\n2. POP");
        printf("\n3. IS FULL?");
        printf("\n4. IS EMPTY?");
        printf("\n5. DISPLAY");
        printf("\n6. EXIT");
        printf("\n\n");
    
        printf("ENTER YOUR CHOICE");
        scanf("%d",&choice);
        int value;

        switch (choice)
        {
        case 1 :printf("\nEnter value to be pushed ");
                scanf("%d",&value);
                push(value);
                break;

        case 2 :value=pop();
                printf("\nValue popped : %d",value);
                break;
        
        case 3 :if (isFull())
                    printf("\nyes");
                else
                    printf("\nno");
                break;
        
        case 4 :if (isEmpty())
                    printf("\nyes");
                else
                    printf("\nno");
                break;
        
        case 5 :display();
                break;
        
        case 6 : exit(0);

        default:printf("\nWrong choice");
        }
    } 
}
